frameworks/Dart/aqueduct/server.dart

```diff
@@ -0,0 +1,228 @@
+class DartAqueductBenchmarkSink extends RequestSink {
+  mustache.Template fortunesTemplate;
+
+  /// Resource initialization code goes here.
+  ///
+  /// Resources like [AuthServer] and [PostgreSQLPersistentStore] should be instantiated
+  /// in this constructor. Configuration behavior - like [HTTPCodecRepository.add] - should be
+  /// configured in this constructor.
+  ///
+  /// The [appConfig] contains configuration data from `aqueduct serve`, e.g.
+  /// the port the application is running on and the path to a configuration file.
+  DartAqueductBenchmarkSink(ApplicationConfiguration appConfig)
+      : super(appConfig) {
+    // Use logging for debuging only
+//    logger.onRecord.listen(
+//        (rec) => print("$rec ${rec.error ?? ""} ${rec.stackTrace ?? ""}"));
+
+    var options =
+        new DartAqueductBenchmarkConfiguration(appConfig.configurationFilePath);
+    ManagedContext.defaultContext = contextWithConnectionInfo(options.database);
+
+    Future.wait([
+      new File('fortunes.mustache').readAsString().then((template) {
+        fortunesTemplate = new mustache.Template(template);
+      })
+    ]);
+  }
+
+  /// All routes must be configured in this method.
+  ///
+  /// This method is invoked after the constructor and before [willOpen].
+  /// All routes must be set up in this method and cannot be added after this method completes.
+  @override
+  void setupRouter(Router router) {
+    router
+        .route("/json")
+        .listen((req) async => new Response.ok({"message": "Hello, World!"})
+          ..contentType = _stripped_json
+          ..headers["date"] = new DateTime.now());
+
+    router
+        .route("/plaintext")
+        .listen((req) async => new Response.ok("Hello, World!")
+          ..contentType = _stripped_text
+          ..headers["date"] = new DateTime.now());
+
+    router.route("/db").listen((req) async {
+      World result = await getRandomWorldObject();
+      return new Response.ok(result)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/queries/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
+            .clamp(_min_query_count, _max_query_count);
+      }
+      List<Future> resultFutures =
+          new List.generate(queryCount, (_) => getRandomWorldObject());
+      List results = await Future.wait(resultFutures);
+      return new Response.ok(results)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/updates/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
```

Comment:
same note here about int.parse failing

Reply:
Requirement is that if it is anything except a valid int it should defaulted to 1 which is done in onError